src/rust/engine/remote_provider/remote_provider_traits/src/lib.rs

```diff
@@ -63,6 +64,10 @@ pub trait ByteStoreProvider: Sync + Send + 'static {
         destination: &mut dyn LoadDestination,
     ) -> Result<bool, String>;
 
+    async fn load_batch(&self, digests: Vec<Digest>) -> Result<Vec<Bytes>, String>;
+
+    fn batch_load_supported(&self) -> bool;
```

Comment:
Idle thought: potentially batch loading is "always" supported by running `load` (to a `Vec<u8>`) concurrently, which is likely no worse than checking `batch_load_supported` and then deciding to call `load` directly.

That is, maybe we could eliminate `batch_load_supported` and implement `load_batch` properly for each provider (given we only have two providers, the exact approach doesn't matter, but if we had more, we might have a default implementation of `load_batch` using `load`).

I think then `batch_load_enabled` doesn't need to be fed all the way down to the `remote_provider_reapi` `ByteStore`, but can instead be kept on `Store` instead.